docs/docs/user/migration-guide/v1-v2.md

```diff
@@ -0,0 +1,177 @@
+# Migrate from V1 to V2
+
+## [Derive component type automatically from class name](https://github.com/bakdata/kpops/pull/309)
+
+KPOps automatically infers the component `type` from the class name. Therefore, the `type` and `schema_type` attributes can be removed from your custom components. By convention the `type` would be the lower, and kebab cased name of the class.
+
+```diff
+class MyCoolStreamApp(StreamsApp):
+-    type = "my-cool-stream-app"
++    ...
+```
+
+Because of this new convention `producer` has been renamed to `producer-app`. This must be addressed in your `pipeline.yaml` or `defaults.yaml`.
```

Comment:
```suggestion
Because of this new convention `producer` has been renamed to `producer-app`. This must be addressed in your `pipeline.yaml` and `defaults.yaml`.
```